frontend/src/employee-frontend/components/child-information/DailyServiceTimesSection.tsx

```diff
@@ -0,0 +1,706 @@
+import ErrorSegment from 'lib-components/atoms/state/ErrorSegment'
+import {
+  isRegular,
+  DailyServiceTimes,
+  isIrregular,
+  RegularDailyServiceTimes,
+  IrregularDailyServiceTimes
+} from '../../types/child'
+import {
+  deleteChildDailyServiceTimes,
+  getChildDailyServiceTimes,
+  putChildDailyServiceTimes
+} from '../../api/child/daily-service-times'
+import { faClock, faPen } from '../../../lib-icons'
+import styled from 'styled-components'
+import InlineButton from '../../../lib-components/atoms/buttons/InlineButton'
+import {
+  FixedSpaceColumn,
+  FixedSpaceRow
+} from 'lib-components/layout/flex-helpers'
+import Button from '../../../lib-components/atoms/buttons/Button'
+import { defaultMargins, Gap } from 'lib-components/white-space'
+import Radio from '../../../lib-components/atoms/form/Radio'
+import InputField from '../../../lib-components/atoms/form/InputField'
+import Checkbox from '../../../lib-components/atoms/form/Checkbox'
+import { UIContext } from '../../state/ui'
+import { RequireRole } from '../../utils/roles'
+
+interface Props {
+  id: UUID
+  open: boolean
+}
+
+interface TimeInputRange {
+  start: string
+  end: string
+}
+
+interface SelectableTimeInputRange extends TimeInputRange {
+  selected: boolean
+}
+
+interface FormData {
+  type: 'REGULAR' | 'IRREGULAR' | 'NOT_SET'
+  regular: TimeInputRange
+  monday: SelectableTimeInputRange
+  tuesday: SelectableTimeInputRange
+  wednesday: SelectableTimeInputRange
+  thursday: SelectableTimeInputRange
+  friday: SelectableTimeInputRange
+}
+
+const emptyForm: FormData = {
+  type: 'NOT_SET',
+  regular: {
+    start: '',
+    end: ''
+  },
+  monday: {
+    selected: false,
+    start: '',
+    end: ''
+  },
+  tuesday: {
+    selected: false,
+    start: '',
+    end: ''
+  },
+  wednesday: {
+    selected: false,
+    start: '',
+    end: ''
+  },
+  thursday: {
+    selected: false,
+    start: '',
+    end: ''
+  },
+  friday: {
+    selected: false,
+    start: '',
+    end: ''
+  }
+}
+
+interface ValidationResult {
+  regular: {
+    start: string | null
+    end: string | null
+  }
+  monday: {
+    start: string | null
+    end: string | null
+  }
+  tuesday: {
+    start: string | null
+    end: string | null
+  }
+  wednesday: {
+    start: string | null
+    end: string | null
+  }
+  thursday: {
+    start: string | null
+    end: string | null
+  }
+  friday: {
+    start: string | null
+    end: string | null
+  }
+}
+
+const weekdays = [
+  'monday',
+  'tuesday',
+  'wednesday',
+  'thursday',
+  'friday'
+] as const
+
+const DailyServiceTimesSection = React.memo(function DailyServiceTimesSection({
+  id,
+  open
+}: Props) {
+  const { i18n } = useTranslation()
+  const { setErrorMessage } = useContext(UIContext)
+
+  const [apiData, setApiData] = useState<Result<DailyServiceTimes | null>>(
+    Loading.of()
+  )
+  const [formData, setFormData] = useState<FormData | null>(null)
+  const [editing, setEditing] = useState(false)
```

Comment:
Äkkiä katottuna näyttäis siltä että `editing` on aina päällä kun `formData !== null`, joten onko `editing`-tila turha?

Reply:
Hmm nykytoteutuksessa ei oo turha, koska kun joko klikkaat peruuta tai submittaat formin niin se asettaa `editing = false` jolloin palataan normi näkymään, mutta äkkiä ajateltuna voisihan sitä sen sijaan vaan asettaa `formData = null` ja käyttää sitä pelkästään flaginä. Ainoa mitä mietin on, että onko se luettavuudeltaan intuitiivista että formDatan olemassaolo toimii editing flaginä vai tarviisko se sit esim nimetä jotenkin eri tavalla. Mitä mieltä?

Reply:
`formData` vois melkein olla nimeltään `editing` ehkä?